File: languages/typescript/runtime/src/testing/SyntheticData.ts
/**
 * Synthetic wire bodies for the mock transport. SYNTHETIC DATA ONLY (ADR-SDK-020):
 * every identifier is an obvious placeholder and no real PAN/CVV/PII ever appears here.
 */

export const DEFAULT_MERCHANT_TRANSACTION_ID = 'mtx-synthetic-1';
export const DEFAULT_TRANSACTION_ID = 'txn-synthetic-1';
export const DEFAULT_CORRELATION_ID = 'corr-synthetic-1';

/** A terminal transaction record (`transactionStatus` 1=approved, 2=declined, 3=error). */
export function transaction(
    transactionStatus: number,
    merchantTransactionId: string = DEFAULT_MERCHANT_TRANSACTION_ID,
): string {
    return JSON.stringify({
        transactionId: DEFAULT_TRANSACTION_ID,
        transactionDate: '2026-01-01T00:00:00Z',
        transactionStatus,
        message: 'synthetic outcome',
        responseCode: '00',
        transactionType: 'charge',
        merchantTransactionId,
        currency: 'USD',
        amount: 1999,
    });
}

/** A pending intent record (post-P-2 shape; `state` is the discriminator). */
export function pending(merchantTransactionId: string = DEFAULT_MERCHANT_TRANSACTION_ID): string {
    return JSON.stringify({
        state: 'pending',
        merchantTransactionId,
        receivedAt: '2026-01-01T00:00:00Z',
    });
}

/** A grouped envelope (`includeAllTransactions` shape). */
export function transactionGroup(merchantTransactionId: string = DEFAULT_MERCHANT_TRANSACTION_ID): string {
    return JSON.stringify({
        transactions: [JSON.parse(transaction(1, merchantTransactionId))],
    });
}

/** An ErrorResponse body; `code` is omitted when absent (matching the wire contract). */
export function errorBody(error: string, code?: string): string {
    return JSON.stringify(code === undefined ? { error } : { error, code });
}

File: languages/python/runtime/revaly_sdk/testing/synthetic_data.py
"""Synthetic response bodies for the mock transport (DX contract §d).

SYNTHETIC DATA ONLY (ADR-SDK-020): every identifier below is a fabricated token —
no real PAN, customer, or merchant material may ever appear here.
"""

from __future__ import annotations

import json
from typing import Any, Dict, Optional

DEFAULT_CORRELATION_ID = "corr-synthetic-0001"
DEFAULT_MERCHANT_TRANSACTION_ID = "mtx_synthetic_0001"
DEFAULT_TRANSACTION_ID = "06SYNTHETIC00000000000000001"


def transaction_dict(transaction_status: int) -> Dict[str, Any]:
    """A terminal transaction record with the given ``transactionStatus``."""
    return {
        "transactionId": DEFAULT_TRANSACTION_ID,
        "transactionDate": "2026-01-01T00:00:00Z",
        "transactionStatus": transaction_status,
        "transactionType": "Charge",
        "merchantTransactionId": DEFAULT_MERCHANT_TRANSACTION_ID,
        "message": "synthetic outcome",
        "responseCode": "00",
        "currency": "USD",
        "amount": 2500,
        "gatewayType": "synthetic-gateway",
        "gatewayTransactionId": "gw_synthetic_0001",
    }


def transaction(transaction_status: int) -> str:
    return json.dumps(transaction_dict(transaction_status))


def transaction_group() -> str:
    """The grouped envelope shape (`transactions` list)."""
    return json.dumps({"transactions": [transaction_dict(1)]})


def pending() -> str:
    """The post-P-2 pending intent shape (`state` discriminator)."""
    return json.dumps(
        {
            "state": "pending",
            "merchantTransactionId": DEFAULT_MERCHANT_TRANSACTION_ID,
            "transactionType": "Charge",
            "receivedAt": "2026-01-01T00:00:00Z",
        }
    )


def error_body(message: str, code: Optional[str] = None) -> str:
    body: Dict[str, Any] = {"error": message}
    if code is not None:
        body["code"] = code
    return json.dumps(body)
